src/Models/style_cache.jl

```diff
@@ -0,0 +1,9 @@
+
+"""
+Supertype for all model cache styles.
+"""
+abstract type AbstractCacheStyle <: AbstractModelStyle end
+
+struct CacheEmpty <: AbstractCacheStyle end
+struct CachePressure <: AbstractCacheStyle end
+struct CacheZeroMomentMicro <: AbstractCacheStyle end
```

Comment:
Can we do something similar as the syle_base, e.g., adding something like
```
Models.cache_variable_names(::CacheEmpty) = Nothing
Models.cache_variable_names(::CachePressure) = (:p)
Models.cache_variable_names(::CacheZeroMomentMicro) = (:p, :Φ, :K, P, :e_int, :microphysics_cache)
```
and the corresponding `variable_types` and `variable_spaces` functions.

By doing this, the `default_ode_cache` does not need to be dispatched on different cache styles, but it can automatically extracts the cache varnames and their vartype/spaces, and then loop over, just as done [here](https://github.com/CliMA/ClimaAtmos.jl/blob/main/src/Models/Nonhydrostatic3DModels/nonhydrostatic_3d_model.jl#L34-L41) in the default_init_conditions.

Reply:
This way, when someone needs to add his/her own cache setups, he/she only needs to change the style file and does not have to touch the model file.

Reply:
The `microphysics_cache` is a tuple of variables. I did not know how to loop over it in the way the stat vars are initialised. Can you help me with this?

Reply:
Could we dispatch on ModelStyles here? Like BaseModelStyles to get the cache variable names?